message/message-util/util/src/test/java/org/sakaiproject/message/util/BaseMessageTest.java

```diff
@@ -0,0 +1,300 @@
+import org.sakaiproject.entity.api.Edit;
+import org.sakaiproject.entity.api.Entity;
+import org.sakaiproject.entity.api.EntityManager;
+import org.sakaiproject.entity.api.EntityPropertyNotDefinedException;
+import org.sakaiproject.entity.api.EntityPropertyTypeException;
+import org.sakaiproject.entity.api.Reference;
+import org.sakaiproject.entity.api.ResourceProperties;
+import org.sakaiproject.message.api.Message;
+import org.sakaiproject.message.api.MessageHeader;
+import org.sakaiproject.message.api.MessageHeaderEdit;
+import org.sakaiproject.message.api.MessageService;
+import org.sakaiproject.time.api.Time;
+import org.w3c.dom.Element;
+
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.Silent.class)
+public class BaseMessageTest {
+
+    @Mock
+    private EntityManager entityManager;
+
+    @Mock
+    private Message message;
+
+    @Mock
+    private Reference reference;
+
+    @Mock
+    private MessageHeader messageHeader;
+
+    @Mock
+    private ResourceProperties messageProperties;
+
+    private BaseMessageTestImpl baseMessage;
+
+    @Before
+    public void setUp() {
+        baseMessage = new BaseMessageTestImpl();
+        baseMessage.setEntityManager(entityManager);
+
+        // Set up common mocks
+        when(message.getReference()).thenReturn("message-reference");
+        when(entityManager.newReference("message-reference")).thenReturn(reference);
+        when(reference.getContext()).thenReturn("site-id");
+        when(message.getHeader()).thenReturn(messageHeader);
+        when(message.getProperties()).thenReturn(messageProperties);
+    }
+
+    @Test
+    public void testIsMessageViewableDraft() {
+        // Setup a draft message
+        when(messageHeader.getDraft()).thenReturn(true);
+
+        // A draft message should not be viewable
+        assertFalse("Draft messages should not be viewable", baseMessage.isMessageViewable(message));
+    }
+
+    @Test
+    public void testIsMessageViewableNoDates() throws EntityPropertyNotDefinedException, EntityPropertyTypeException {
+        // Setup a non-draft message with no release or retract dates
+        when(messageHeader.getDraft()).thenReturn(false);
+        when(messageProperties.getInstantProperty(MessageService.RELEASE_DATE))
+            .thenThrow(EntityPropertyNotDefinedException.class);
+        when(messageProperties.getInstantProperty(MessageService.RETRACT_DATE))
+            .thenThrow(EntityPropertyNotDefinedException.class);
+
+        // A message with no dates should be viewable
+        assertTrue("Message with no dates should be viewable", baseMessage.isMessageViewable(message));
+    }
+
+    @Test
+    public void testIsMessageViewableFutureReleaseDate() throws EntityPropertyNotDefinedException, EntityPropertyTypeException {
```

Comment:
Should we really not see announcements that are going to be released in the Future as an instructor?

Reply:
you should view them in the announcements tool. you should not see them in the Overview or Dashboard tools.